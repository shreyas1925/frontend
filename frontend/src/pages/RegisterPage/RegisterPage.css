*{
    margin:0;
    padding:0;
}
.main--container{
    width: 100vw;
    height: 100vh;
    display: flex;
}


.right--container{
    width: 30%;
    height: 100%;
    background-color: #272727;
    display: flex;
    justify-content: center;
    align-items: center;
    flex-direction: column;
}

.right--container h1{
    color:white;
    font-family: "Roboto", sans-serif;
    font-size: 25px;
    font-weight: 400;
    margin-bottom: 100px;
}

.form--container{
    display: flex;
    flex-direction: column;
    margin-left: -30px;
}

.form--container input{
    margin: 20px;
    width:100%;
    height: 20px;
    padding: 8px;
    border-radius: 8px;
    outline: none;
}

button{
    margin: 22px;
    width:110%;
    height: 38px;
    border-radius: 8px;
    color:white;
    background-color: #5d48f5;
    border: none;
    cursor: pointer;
}

p{
    color: #7d7d7d;
    margin-left: 22px;
}

.form--footer{
    display: flex;

}

.form--footer a{
    color: #7d7d7d;
    text-decoration: none;
    margin-left: 25px;
    cursor: pointer;
}

.left--container{
    width: 70%;
    height: 100%;
    background-color: #f0f2ff;
    display: flex;
    flex-direction: column;
}

.image--container img{
    height: 400px;
    width: 450px;
    margin-left:280px;
    margin-top: -50px;
    background-size: cover;
    background-position: center;
    background-repeat: no-repeat;
}

.text--container h1{
    color:black;
    padding: 0px;
    margin:12px;
    margin-left: 60px;
    font-weight: 800;
    font-family: "Roboto", sans-serif;
}

.text--container{
    color: #7d7d7d;
    margin-top: 100px;
    margin-left: 100px;
}

@media screen and (max-width: 768px){
    .left--container{
        display: none;
    }
    .right--container{
        width: 100%;
    }
}