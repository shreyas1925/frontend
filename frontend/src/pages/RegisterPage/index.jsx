import React from "react";
import { Form } from "../../components";
import axios from "axios";
import { useFormik } from "formik";
import "./RegisterPage.css"
import people from "../../assets/people.png"
import { useNavigate } from "react-router-dom";

const RegisterPage = () => {
    const navigate = useNavigate();
    const initialValues = {
        email: '',
        password:''
    };

    const { values, errors, handleChange, handleBlur, touched, handleSubmit } =
    useFormik({
        initialValues: initialValues,
        // validationSchema: UserSchema,
        onSubmit: (values,action) => {
            console.log(values);
            axios({
                method: 'POST',
                url: 'http://localhost:4000/api/register',
                data: values
            })
                .then(function (res) {
                    console.log(res);
                    alert('Successfully registered!');  
                })
                .catch(function (res) {
                    console.log(res);
                });
        
            action.resetForm();
            navigate("/login")
        },
      
    });

  return (
    <div className="main--container">
      <div className="left--container">
        <div className="text--container">
            <h1>Design APIs fast,
            </h1>
            <h1>
            Manage content easily</h1>
        </div>
        <div className="image--container">
            <img src={people} alt="" />
        </div>
      </div>
      <div className='right--container'>
        <h1>Register to your CMS+ account</h1>
       <form action="" className="form--container" onSubmit={handleSubmit}>
        <p>Email</p>
        <input
          autoComplete="off"
          type="email"
          name="email"
          value={values.email}
          onChange={handleChange}
          onBlur={handleBlur}
        />
        {errors.email && touched.email ? (
          <p className="form--error">{errors.email}</p>
        ) : null}
        
        <p>Password</p>
        <input
          autoComplete="off"
          type="password"
          name="password"
          value={values.password}
          onChange={handleChange}
          onBlur={handleBlur}
        />
        {errors.password && touched.password ? (
          <p className="form--error">{errors.password}</p>
        ) : null}

        <button type="submit" className="btn">Register</button>

        <div className="form--footer">
        <p>Already an user ? </p> <a href="/login">Login</a>
        </div>
      </form>
    </div>
    </div>
  );
};

export default RegisterPage;
