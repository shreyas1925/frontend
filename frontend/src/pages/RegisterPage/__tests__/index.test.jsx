import RegisterPage from "..";
import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
describe("RegisterPage", () => {
  it("should render correctly", () => {
    const { asFragment } = render(
      <BrowserRouter>
        <RegisterPage />
      </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it("should render and have text", () => {
    const { getByText } = render(
      <BrowserRouter>
        <RegisterPage />
      </BrowserRouter>
    );
    expect(getByText("Register")).toBeInTheDocument();
  });
});
