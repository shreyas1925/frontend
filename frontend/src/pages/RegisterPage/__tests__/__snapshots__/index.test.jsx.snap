// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterPage should render correctly 1`] = `
<DocumentFragment>
  <div
    class="main--container"
  >
    <div
      class="left--container"
    >
      <div
        class="text--container"
      >
        <h1>
          Design APIs fast,
        </h1>
        <h1>
          Manage content easily
        </h1>
      </div>
      <div
        class="image--container"
      >
        <img
          alt=""
          src="people.png"
        />
      </div>
    </div>
    <div
      class="right--container"
    >
      <h1>
        Register to your CMS+ account
      </h1>
      <form
        action=""
        class="form--container"
      >
        <p>
          Email
        </p>
        <input
          autocomplete="off"
          name="email"
          type="email"
          value=""
        />
        <p>
          Password
        </p>
        <input
          autocomplete="off"
          name="password"
          type="password"
          value=""
        />
        <button
          class="btn"
          type="submit"
        >
          Register
        </button>
        <div
          class="form--footer"
        >
          <p>
            Already an user ? 
          </p>
           
          <a
            href="/login"
          >
            Login
          </a>
        </div>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
